src/Type/EntityQuery/EntityQueryType.php

```diff
@@ -17,9 +17,17 @@ public function hasAccessCheck(): bool
 
     public function withAccessCheck(): self
     {
-        $type = clone $this;
+        $type = new self(
+            $this->getClassName(),
+            $this->getSubtractedType(),
+            $this->getClassReflection()
+        );
         $type->hasAccessCheck = true;
```

Comment:
cc @brambaud since you contributed this code. I didn't think twice when we first added this, but cloning was copying over cached information, leading to the bugs.

Reply:
:/ looks like this caused more failures although it fixed the one bug. Probably because I used `self` instead of `static`.

Reply:
Hooo! 😮 
Nice catch! Didn't think of that! 🙇